crates/csharp/src/function.rs

```diff
@@ -1375,15 +1401,27 @@ impl Bindgen for FunctionBindgen<'_, '_> {
 
             Instruction::FutureLower { .. } => {
                 let op = &operands[0];
+                self.interface_gen.add_future(self.func_name);
+
                 results.push(format!("{op}.Handle"));
```

Comment:
`FutureReader` should probably have a `TakeHandle()` method that zeros out the handle field (and asserts that it wasn't already zero) before returning the original value so that the application won't accidentally try to use the no-longer-valid handle.

Reply:
I've done this, but not sure I understand in what situation it would be used.